easybuild/easyconfigs/r/RELION/RELION-3.1.3-fosscuda-2020b.eb

```diff
@@ -0,0 +1,48 @@
+# This file is an EasyBuild reciPY as per https://github.com/easybuilders/easybuild
+# modified from https://github.com/FredHutch/easybuild-life-sciences/blob/main/fh_easyconfigs/r/RELION/RELION-3.1.2-fosscuda-2020b.eb
+
+easyblock = 'CMakeMake'
+
+name = 'RELION'
+version = '3.1.3'
+
+github_account = '3dem'
+homepage = 'https://github.com/%(github_account)s/%(namelower)s'
+description = """RELION (for REgularised LIkelihood OptimisatioN) is a stand-alone computer
+ program for Maximum A Posteriori refinement of (multiple) 3D reconstructions or 2D class
+ averages in cryo-electron microscopy."""
+
+toolchain = {'name': 'fosscuda', 'version': '2020b'}
+toolchainopts = { 'openmp' : True }
+
+source_urls = ['https://github.com/%(github_account)s/%(namelower)s/archive']
+sources = ['%(version)s.tar.gz']
+checksums = ['f10934058e0670807b7004899d7b5103c21ad77841219352c66a33f98586d42e']
+
+builddependencies = [
+    ('CMake', '3.18.4'),
+]
+
+dependencies = [
+    ('X11', '20201008'),
+    ('LibTIFF', '4.1.0'),
+    ('FLTK', '1.3.5'),
+    ('ctffind', '4.1.14'),
+    ('tbb', '2020.3'),
+    ('Ghostscript', '9.53.3')
+]
+
+hiddendependencies = [
+    ('LibTIFF', '4.1.0'),
+]
+
+configopts = '-DCUDA_ARCH=60 '
```

Comment:
@aseedb Should this be hardcoded to this specific CUDA compute capability, or would it be better to have this easily configurable?

EasyBuild already has a global configuration option for this: `--cuda-compute-capabilities`, and corresponding template values like `%(cuda_cc_space_sep)s` are set based on that.

Could that be leveraged here?

Reply:
@aseedb You're now not specifying any configure option at all anymore to specify for which CUDA target `RELION` should be built, so RELION is probably falling back to some default?

Reply:
@boegel Oh yes, I see! But I have a problem. RELION wants the '-DCUDA_ARCH=ab ' where a and b are two digits.
So for example if you want to use compute capability 6.0, you need to have '-DCUDA_ARCH=60'.
--cuda-compute-capabilities requires input in this format: 'a.b' 

I have tried to do something like that in the config file
```
cuda_compute_capabilities = ['6.0']
local_caps = [''.join(x.split('.')) for x in %(cuda_compute_capabilities)s]
local_arch = local_caps[0]

configopts = '-DCUDA_ARCH=' + local_arch + ' '

```

This partially works. It compiles with -DCUDA_ARCH=60.
However, whatever I input via the command line --cuda-compute-capabilities has no effect.
And if I comment out `cuda_compute_capabilities = ['6.0']` then an invalid syntax is thrown. 

Can you guide me on how to solve my problem?